pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -705,6 +718,10 @@ stdenv.mkDerivation ({
 // optionalAttrs (args ? preFixup)               { inherit preFixup; }
 // optionalAttrs (args ? postFixup)              { inherit postFixup; }
 // optionalAttrs (args ? dontStrip)              { inherit dontStrip; }
+// optionalAttrs enableSeparateDistOutput {
+     # Don't strip the dist output
+     preFixupHooks = [ "outputs=(\${outputs[@]/dist})" ];
```

Comment:
I'm purely just curious, but how does this prevent the `dist` output from getting stripped?  (I'm not familiar with `preFixupHooks`.)